watcher.go

```diff
@@ -109,7 +109,7 @@ func (w *Watcher) Update() error {
 	w.lock.Lock()
 	defer w.lock.Unlock()
 	resp, err := w.client.Put(context.TODO(), w.keyName, "")
-	if err != nil {
+	if err == nil {
```

Comment:
Why do this change?

Reply:
When policy changes, etcd-watcher will `Put` a message into etcd, then the revision of key will be updated. 
```go
resp, err := w.client.Put(context.TODO(), w.keyName, "")
```
At the same time, etcd-watcher will get this new revision because it is watching this key.
```go
func (w *Watcher) startWatch() error {
	watcher := w.client.Watch(context.Background(), w.keyName)
```

So in last pr #22 , to prevent consuming the messages put by self, every time etcd-watcher put a message, it will record the revision from the `Put` response. And in the goroutine for `Watch`, it will verify the revision of received message is bigger than recorded one or not.
```go
for res := range watcher {
    t := res.Events[0]
    if t.IsCreate() || t.IsModify() {
	  w.lock.RLock()
	  //ignore self update
	  if rev := t.Kv.ModRevision; rev > w.lastSentRev && w.callback != nil {
		  w.callback(strconv.FormatInt(rev, 10))
	  }
	  w.lock.RUnlock()
    }
}
```

But in the `Update` function, it only update local recorded revision when `Put` fails, it means this feature will be never triggered.
```go
func (w *Watcher) Update() error {
	w.lock.Lock()
	defer w.lock.Unlock()
	resp, err := w.client.Put(context.TODO(), w.keyName, "")
        // We should update lastSentRev when put operation succeeds, but not when it fails.
        // if err != nil {
	if err == nil {
		w.lastSentRev = resp.Header.GetRevision()
	}
	return err
}
```